diff --git a/cytoolz/itertoolz.pyx b/cytoolz/itertoolz.pyx
index 6c5bc88..b3b026e 100644
--- a/cytoolz/itertoolz.pyx
+++ b/cytoolz/itertoolz.pyx
@@ -963,6 +963,7 @@ cdef class sliding_window:
         cdef Py_ssize_t i
         self.iterseq = iter(seq)
         self.prev = PyTuple_New(n)
+        Py_INCREF(None)
         PyTuple_SET_ITEM(self.prev, 0, None)
         for i, seq in enumerate(islice(self.iterseq, n-1), 1):
             Py_INCREF(seq)
@@ -1056,6 +1057,7 @@ cdef class partition_all:
         if i == 0:
             raise StopIteration
         for j in range(i, self.n):
+            Py_INCREF(None)
             PyTuple_SET_ITEM(result, j, None)
         return PyTuple_GetSlice(result, 0, i)
 
